repo/buildrepo/build.go

```diff
@@ -86,14 +86,29 @@ func New(ctx context.Context, path string, cfg *config.Config) (repo.Repo, error
 	}
 }
 
+var fsMap = map[string]qfs.Filesystem{
+	"local": localfs.NewFS(),
+	"http":  httpfs.NewFS(),
+}
+
 // NewFilesystem creates a qfs.Filesystem from configuration
 func NewFilesystem(cfg *config.Config, store cafs.Filestore) (qfs.Filesystem, error) {
-	mux := map[string]qfs.Filesystem{
-		"local": localfs.NewFS(),
-		"http":  httpfs.NewFS(),
-		"cafs":  store,
+	mux := map[string]qfs.Filesystem{}
+
+	if cfg != nil && cfg.Filesystems != nil && len(cfg.Filesystems) > 0 {
+		for _, fsCfg := range cfg.Filesystems {
+			if fs, ok := fsMap[fsCfg.Type]; ok {
+				mux[fsCfg.Type] = fs
```

Comment:
Need to work out how to pass the actual config into the constructors. Feels like `MuxConfig` for `Filesystem` is the way to go and then I can simply replicate this behavior https://github.com/qri-io/qfs/blob/8af27780e44bbeefa98491c7146450a256b246b9/muxfs/mux.go#L79

Reply:
This file gets decimated when we upgrade to the latest qfs, we no longer rely on `buildrepo` and almost solely on the `muxfs` package to add a filesystem to a qri instance. Again, as long as tests are passing, I think this should be fine!